### PR TITLE
Add 1.12 release notes. Fixes #18939.

### PR DESCRIPTION
**Pull request by jacobroufa**

### CODE REVIEW

File: releasenotes/1.12.rst [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+* Chrome 13-50
+
+* IE 8-11
+
+* Opera 10.50-12 (Dojo core only). Blink-based versions of Opera should work like Chrome, but are not officially tested or supported for this release.
+
+**Mobile (dojox/mobile)**
+
+* iOS (Mobile Safari), 6.x, 7.x, 8.x, and 9.x (including all Dijit widgets except Editor)
```
Comment (dylans) [inline releasenotes/1.12.rst]: Add iOS 10.x ?

File: releasenotes/1.12.rst [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+* Android (Android Browser and Chrome) 2.3, 4.0-4.4, 5.0-5.1, and  6.0 (Platform browser and Chrome browser) (including all Dijit widgets for Android 4.0-4.4 except Editor)
```
Comment (dylans) [inline releasenotes/1.12.rst]: Verify this is the most recent Chrome for Android version please.

File: releasenotes/1.12.rst [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+The following user agents have been tested and are supported in this release. If a user agent version is not listed as
+supported, it still may work, especially on browsers with rapid release cycles (Chrome, Firefox). Future minor releases
+of Dojo might include fixes to support future releases of these browsers, but it is not guaranteed.
+
+**Desktop**
+
+* Edge 12
+
+* Firefox 3.6-45
```
Comment (dylans) [inline releasenotes/1.12.rst]: Through Firefox 50?

File: releasenotes/1.12.rst [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+* Safari 6-9
```
Comment (dylans) [inline releasenotes/1.12.rst]: Through Safari 10?

File: releasenotes/1.12.rst [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+
+* Chrome 13-50
```
Comment (dylans) [inline releasenotes/1.12.rst]: Through Chrome 54?

File: releasenotes/1.12.rst [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+* BlackBerry 6-7 & 10 (Platform browser)
+
+* Windows Phone 8 (IE10)
+
+* Mobile compatibility on desktop browsers: Edge 12, IE 8-11, Firefox 4-43, Chrome 13-47
+
+
+**Server-Side User Agents**
+
+* Node 0.10.X, 0.12.x, 4.x, 5.x
```
Comment (dylans) [inline releasenotes/1.12.rst]: 6.x ?

File: releasenotes/1.12.rst [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+
+* iOS (Mobile Safari), 6.x, 7.x, 8.x, and 9.x (including all Dijit widgets except Editor)
+
+* Android (Android Browser and Chrome) 2.3, 4.0-4.4, 5.0-5.1, and  6.0 (Platform browser and Chrome browser) (including all Dijit widgets for Android 4.0-4.4 except Editor)
+
+* BlackBerry 6-7 & 10 (Platform browser)
+
+* Windows Phone 8 (IE10)
+
+* Mobile compatibility on desktop browsers: Edge 12, IE 8-11, Firefox 4-43, Chrome 13-47
```
Comment (dylans) [inline releasenotes/1.12.rst]: Update to match list above please.

File: releasenotes/1.12.rst [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+
+**Server-Side User Agents**
+
+* Node 0.10.X, 0.12.x, 4.x, 5.x
+
+General
+=======
+
+* Update language in documentation to reflect Dojo Foundation being superseded by the JS Foundation.
+* Updated copyright year in README for Dojo, Dijit, DojoX, Util.
```
Comment (dylans) [inline releasenotes/1.12.rst]: Probably not worth mentioning a copyright year update :)

File: releasenotes/1.12.rst [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+* Aligned selected and unselected checkboxes in dojox/mobile in Edge.
+* Fixed horizontal dojox/mobile/Slider event propogation, resulting in becoming unusable in a horizontal dojox/mobile/ScrollablePane.
+
+Util
+====
+
+Build System
+------------
+
+* Updated Closure Compiler to v20160713. Added ``compilationLevel`` option to ``optimizeOptions``. Added all compilation level set functions to ``optimizeOptions``. Added ``languageOut`` option to ``optimizeOptions``. Updated ``languageIn`` and ``languageOut`` options to use ``fromString`` enum function. Updated documentation accordingly.
```
Comment (dylans) [inline releasenotes/1.12.rst]: Explain that this adds support for ES6 grammar?

File: releasenotes/index.rst [hunk 10]
```diff
@@ -13,16 +13,17 @@ see the Release Notes for it.
 Current Stable
 ==============
 
-* :ref:`Dojo Toolkit v1.10 Release Notes <releasenotes/1.10>`
+* :ref:`Dojo Toolkit v1.11 Release Notes <releasenotes/1.11>`
```
Comment (dylans) [inline releasenotes/index.rst]: Should be 1.12

File: releasenotes/index.rst [hunk 11]
```diff
@@ -13,16 +13,17 @@ see the Release Notes for it.
...
 
 Current Development
 ===================
 
-* :ref:`Dojo Toolkit v1.10 Release Notes <releasenotes/1.11>`
+* :ref:`Dojo Toolkit v1.12 Release Notes <releasenotes/1.12>`
```
Comment (dylans) [inline releasenotes/index.rst]: Should be 1.13

File: releasenotes/index.rst [hunk 12]
```diff
@@ -13,16 +13,17 @@ see the Release Notes for it.
...
 
 Archive
 =======
 
+* :ref:`Dojo Toolkit v1.10 Release Notes <releasenotes/1.10>`
```
Comment (dylans) [inline releasenotes/index.rst]: Should add 1.11 here.

**Comment by jacobroufa** [comment]
@dylans please let me know if these changes are acceptable!

File: releasenotes/index.rst [hunk 13]
```diff
@@ -13,16 +13,17 @@ see the Release Notes for it.
...
 ==============
 
-* :ref:`Dojo Toolkit v1.11 Release Notes <releasenotes/1.11>`
+* :ref:`Dojo Toolkit v1.12 Release Notes <releasenotes/1.12>`
 
 Current Development
 ===================
 
-* :ref:`Dojo Toolkit v1.12 Release Notes <releasenotes/1.12>`
+* :ref:`Dojo Toolkit v1.13 Release Notes <releasenotes/1.13>`
```
Comment (dylans) [inline releasenotes/index.rst]: This looks good, let's remember to add this file also and then we're good to go.

Comment (jacobroufa) [inline releasenotes/index.rst]: Added `1.13.rst` with ToC and sections as taken from `1.12.rst` and prior notes.

**Comment by sindilevich** [comment]
Hello @jacobroufa,

The Google Closure Compiler in dojo 1.12.1 is _v20160911_, and not v20160713. See https://github.com/dojo/util/commit/f8656829ad3a84b2e41f901758dd83532db80a8e for details.

**Comment by jacobroufa** [comment]
@sindilevich Please see #148 for the resolution to that. The version number I pulled was from the original ticket to update Closure Compiler, not the later one that fixed a different issue with the updated compiler. Good catch!

**Pull request merged**